pkg/controller/deployment/recreate.go

```diff
@@ -31,7 +31,7 @@ func (dc *DeploymentController) rolloutRecreate(d *apps.Deployment, rsList []*ap
 	if err != nil {
 		return err
 	}
-	allRSs := append(oldRSs, newRS)
+	allRSs := append(append([]*apps.ReplicaSet{nil}, oldRSs...), newRS)
```

Comment:
`[]*apps.ReplicaSet{nil}` will add a nil element that we don't expected to the slice.
Same as below.

Reply:
done